### PR TITLE
Poc tetris engine

### PR DESCRIPTION
**Pull request by rizky**

### CODE REVIEW

File: src/client/src/models/block.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import _ from 'lodash';
+
+import { blockShape, origin } from '/constants/tetriminos.ts';
+
+class Block {
```
Comment (rizky) [inline src/client/src/models/block.ts:5]: I know we should not use Class.
Just POC, will convert this to function.
It's so clean with class tho

File: src/client/src/models/block.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  xy: number[];
```
Comment (rizky) [inline src/client/src/models/block.ts:6]: probably will split this to row and col to make more readable

File: src/client/src/screens/Playground/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import * as React from 'react';
+import useInterval from '@use-it/interval';
+import useKey from 'use-key-hook';
+
+import Matrix from '/components/Matrix';
+import { blankMatrix } from '/constants/tetriminos';
+import Block from '/models/block';
+import { blockType } from '/constants/tetriminos';
+
+const printBlock = (matrix: Matrix, block: Block) => {
```
Comment (rizky) [inline src/client/src/screens/Playground/index.tsx:12]: put to service and make sure it's immutable

File: src/client/src/constants/tetriminos.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  O: [
+    [1, 1],
+    [1, 1],
+  ],
+  T: [
+    [0, 1, 0],
+    [1, 1, 1],
+  ],
+};
```
Comment (mrakhman) [inline src/client/src/constants/tetriminos.ts:31]: L == J
Z == S

Right? They are just rotations of each other

image

**Pull request merged**